test/test_item_model_utils.py

```diff
@@ -0,0 +1,388 @@
+        result = portal_item._undo_make_hashable_auth(hashable_auth)
+        assert result == expected
+
+    @pytest.mark.parametrize(
+        "raise_exception,expected", [(False, SOME_ITEM_PROPERTIES), (True, {})]
+    )
+    def test_get_and_cache_item_via_auth(
+        self, raise_exception: bool, expected: Dict[str, Any]
+    ) -> None:
+        side_effect = Exception if raise_exception else None
+        random_add_on = str(random())  # To differentiate parametrized calls
+        with patch_get_metadata(
+            side_effect=side_effect, return_value=SOME_ITEM_PROPERTIES
+        ) as mock_get_metadata:
+            portal_item = get_portal_item()
+            result = portal_item._get_and_cache_item_via_auth(
+                SOME_UUID, HASHABLE_SOME_AUTH, add_on=random_add_on
+            )
+            assert result == expected
+            assert len(mock_get_metadata.call_args_list) == 1
+            mock_get_metadata.assert_called_once_with(
+                SOME_UUID, key=SOME_AUTH, add_on=random_add_on
+            )
+
+            # Ensure cached
+            portal_item._get_and_cache_item_via_auth(
+                SOME_UUID, HASHABLE_SOME_AUTH, add_on=random_add_on
+            )
+            assert len(mock_get_metadata.call_args_list) == 1
+
+    @pytest.mark.parametrize(
+        "auth,fetch_links,exception_expected",
+        [
+            (None, False, True),
+            (None, True, True),
+            (SOME_AUTH, False, False),
+            (SOME_AUTH, True, False),
+        ],
+    )
+    def test_from_identifier_and_existing_item(
+        self, auth: Dict[str, Any], fetch_links: bool, exception_expected: bool
+    ) -> None:
+        identifier = SOME_UUID
+        portal_item = get_portal_item(
+            auth=auth, fetch_links=fetch_links, use_defaults=False
+        )
+        with patch_from_identifier_and_auth() as mock_from_identifier_and_auth:
+            if exception_expected:
+                with pytest.raises(RuntimeError):
+                    PortalItem.from_identifier_and_existing_item(
+                        identifier, portal_item
+                    )
+            else:
+                result = PortalItem.from_identifier_and_existing_item(
+                    identifier, portal_item
+                )
+                assert result == mock_from_identifier_and_auth.return_value
+                mock_from_identifier_and_auth.assert_called_once_with(
+                    identifier, auth, fetch_links=fetch_links
+                )
+
+    def test_from_properties_and_existing_item(self) -> None:
+        properties = OTHER_ITEM_PROPERTIES
+        portal_item = get_portal_item()
+        result = PortalItem.from_properties_and_existing_item(properties, portal_item)
+        assert isinstance(result, PortalItem)
+        assert result.get_auth() == portal_item.get_auth()
+        assert result.should_fetch_links() == portal_item.should_fetch_links()
+
+
+def get_nested_property(
```

Comment:
I assume this is just making a sample property. I find this kind of thing hard to read because in test files, it's often hard to tell what the newly-defined functions to be tested are and what are the testing artifacts. I like to name things like this something like `make_sample_tested_property` so that in context of use, it's clear it's a testing artifact. Also, and it's a subtle point, but I like verbs like `get`, `find`, `lookup`, and `make` to have consistent meanings, and while we don't have a global convention for some of the subtleties I use for myself, I do think `get` usually connotes finding something that exists already, and here you're making something anew, which is why I'd use `make` as the part of the function name that's the verb, though maybe you're thinking of this as a representation of something that exists in the data? If so, I would leave it as `get` but I would clarify that in a doc string, which should be present in any case since this function does something non-trivial. 

Note that I'm not perfect at writing doc strings myself, but Will usually beats me up about it, presumably because it's hard to look at someone else's code and figure out what they're doing and he's usually on the outside of what I'm doing trying to claw his way in, as I'm doing here. :)

Reply:
On the subject of docstrings, I really think some of these tests could use them and even some inline comments. They are quite complex relatively speaking and merit such documentation.